database-mongodb/Riak/seed.js

```diff
@@ -0,0 +1,135 @@
+      headers: {
+        'Authorization': `Client-ID ${token}`,
+        'Accept-Version': 'v1'
+      }
+    };
+
+    promisesArray.push(axios(options));
+  }
+
+  for (let i = 1; i <= numberOfRequests; i++) {
+    let options = {
+      method: 'get',
+      url: `https://api.unsplash.com/search/photos?query=kitten&page=${i}&per_page=${urlsPerRequest}`,
+      headers: {
+        'Authorization': `Client-ID ${token}`,
+        'Accept-Version': 'v1'
+      }
+    };
+
+    promisesArray.push(axios(options));
+  }
+
+  return Promise.all(promisesArray)
+    .then(extractURLs);
+};
+
+const groupImageData = (urls, batch) => {
+  const batchOfItemObjects = [];
+  const startingItemIdForThisBatch = 1000 * batch + 100;
+  const amountOfObjectItemsToBeRandomlyGenerated = 1000;
+  let count = 0;
+
+  while (count < amountOfObjectItemsToBeRandomlyGenerated) {
+    const catOrDog = Math.floor(Math.random() * 2);
+    let startIndex = 0;
+    let endIndex = urls.length / 2;
+
+    if (catOrDog) {
+      startIndex = endIndex;
+      endIndex = urls.length;
+    }
+
+
+    let numberOfPicturesItemHas = Math.floor(Math.random() * 3) + 1;
+
+    const picturesItemHas = [];
+
+    while (numberOfPicturesItemHas > 0) {
+      const randomPictureIndex = Math.floor(Math.random() * (endIndex - startIndex)) + startIndex;
+      const potentialPicture = urls[randomPictureIndex];
+
+      if (picturesItemHas.indexOf(potentialPicture) === -1) {
+        picturesItemHas.push(potentialPicture);
+        numberOfPicturesItemHas--;
+      }
+    }
+
+    const itemId = (startingItemIdForThisBatch + count).toString();
+    const picturesItemHasString = picturesItemHas.join('XXX');
+
+    batchOfItemObjects.push({ itemId, itemImages: picturesItemHasString });
+
+    count++;
+  }
+
+  return batchOfItemObjects;
+};
+
+const insertImages = function (urls, totalNumberOfBatches, actuallyInsert) {
+  const dataInsertions = [];
+
+  for (let i = 0; i < totalNumberOfBatches; i++) {
+    const batchOfItemObjects = groupImageData(urls, i);
+    if (actuallyInsert) {
+      dataInsertions.push(Images.insertRecords(batchOfItemObjects));
+    } else {
+      dataInsertions.push(batchOfItemObjects);
+    }
+  }
+
+  return Promise.all(dataInsertions);
+};
+
+const handleSeeding = function(numberOfRequests, urlsPerRequest, totalNumberOfBatches, actuallyInsert) {
+
+  return getUnsplashImages(numberOfRequests, urlsPerRequest)
+    .then((urlsArray) => {
+      return insertImages(urlsArray, totalNumberOfBatches, actuallyInsert);
+    })
+    .then(() => {
+      console.log('Successfully inserted images');
+      connect.stop();
+    })
+    .catch((error) => console.log(error)
+    );
+};
+
+module.exports.promisesArray = promisesArray;
+module.exports.extractURLs = extractURLs;
+module.exports.insertImages = insertImages;
+module.exports.getUnsplashImages = getUnsplashImages;
+module.exports.groupImageData = groupImageData;
+module.exports.handleSeeding = handleSeeding;
```

Comment:
Can't you just import functionality instead of copying and pasting it for dryness?

Reply:
Sorry, Eric. I am not sure what this comment is referring to. Could you please clarify?

Reply:
This whole file has a lot of the same functionality of your other seed file. Can't you just import those functions from the other file here? It's not a big deal since you'll be using one database anyways, I was just wondering since you're code is being repeated twice.

Reply:
Oh, I just didn't want to think that far. In the end, i will have to choose a final database and then only one of the three sets of seeding files will remain. So yes, it is duplicate, but it won't be a week from now.

Thanks again and thanks for paying such close attention to detail.